addons/project_hr_expense/models/project.py

```diff
@@ -68,11 +68,21 @@ def _get_profitability_sequence_per_invoice_type(self):
         sequence_per_invoice_type['expenses'] = 11
         return sequence_per_invoice_type
 
+    def _get_already_included_profitability_invoice_line_ids(self):
+        # As both purchase orders and expenses (paid by employee) create vendor bills,
+        # we need to make sure they are exclusive in the profitability report.
+        query = self.env['account.move.line']._search([('move_id.expense_sheet_id', '!=', False)])
+        query.order = None
+        query_string, query_param = query.select('"account_move_line".id')
+        self._cr.execute(query_string, query_param)
+        expense_account_move_line_ids = [r[0] for r in self._cr.fetchall()]
+        return super()._get_already_included_profitability_invoice_line_ids() + expense_account_move_line_ids
```

Comment:
Why not simply do that:

```suggestion
        move_line_ids = super()._get_already_included_profitability_invoice_line_ids()
        query = self.env['account.move.line']._search([
            ('move_id.expense_sheet_id', '!=', False),
            ('id', 'not in', move_line_ids),
        ])
        query.order = None
        return move_line_ids + list(query)
```

🤔 

`('id', 'not in', move_line_ids)` is there to avoid having duplicate ids in the list returned by that method. 🙂

Reply:
thanks! really nice suggestion!